### PR TITLE
[chip_earlgrey,spi_device,sival] Add new metadata to chip testplan for SiVal

### PR DESCRIPTION
**Pull request by hcallahan-lowrisc**
Not all 'features' are covered by the current tests, this may be further fixed by either adding new tests or changing the granularity of the features identified.

### CODE REVIEW

File: hw/top_earlgrey/data/ip/chip_spi_device_testplan.hjson [hunk 1]
```diff
@@ -11,92 +12,144 @@
...
+              model.
             '''
       stage: V3
+      si_stage: None
+      lc_states: [ "PROD" ]
+      features: [
+        "SPI_DEVICE.MODE.PASSTHROUGH",
+        "SPI_DEVICE.HW.LANES",
+        "SPI_DEVICE.MODE.PASSTHROUGH.CMD_FILTER",
+        "",
```
Comment (rswarbrick) [inline hw/top_earlgrey/data/ip/chip_spi_device_testplan.hjson:115]: I don't think this empty string should be here?

File: hw/top_earlgrey/data/ip/chip_spi_device_testplan.hjson [hunk 2]
```diff
@@ -11,92 +12,144 @@
...
       desc: '''Verify the collisions on driving spi_host is handled properly.
 
             - Enable upload related interrupts and configure the spi_device in passthrough mode.
             - Configure a command slot to enable upload for a flash program/erase command.
             - Excecute two parallel threads:
-              1. SPI host agent.
-                - Send this command via an upstream SPI host agent, then the agent keeps sending
-                read_status to poll the busy bit.
+              1. Upstream SPI host agent.
+                - Send a flash program/erase, then keeps sending read_status to poll the busy bit.
```
Comment (rswarbrick) [inline hw/top_earlgrey/data/ip/chip_spi_device_testplan.hjson:128]: Nit (grammar): I think you want "keep sending" here.

File: hw/top_earlgrey/data/ip/chip_spi_device_testplan.hjson [hunk 3]
```diff
@@ -11,92 +12,144 @@
...
                 - When the busy bit is low, issue a read command to read data from the downstream
                   SPI port, and check data correctness.
               2. A SW process.
-                - SW receives an upload interrupt and reads the command in the upload fifo to check.
-                - SW configures the SPI host that shows the same downstream port, to send the
+                - Receives an upload interrupt, then read the command in the upload fifo to check.
```
Comment (rswarbrick) [inline hw/top_earlgrey/data/ip/chip_spi_device_testplan.hjson:132]: Nit (grammar): I think you want "reads" here.

File: hw/top_earlgrey/data/ip/chip_spi_device_testplan.hjson [hunk 4]
```diff
@@ -105,10 +158,16 @@
...
+      lc_states: [ "PROD" ]
+      features: [
+        "SPI_DEVICE.MODE.TPM",
+        "SPI_DEVICE.MODE.TPM.READ_FIFO_MODE",
+      ]
       tests: ["chip_sw_spi_device_tpm"]
     }
-
     {
+      // This test is more a pinmux/padctrl test than a spi_device one. Consider reclassification.
```
Comment (rswarbrick) [inline hw/top_earlgrey/data/ip/chip_spi_device_testplan.hjson:170]: I agree. Would it also make sense (for now) to leave a "Notes for silicon targets" section in the `desc` to explain why `features` is empty?

Comment (hcallahan-lowrisc) [reply hw/top_earlgrey/data/ip/chip_spi_device_testplan.hjson:171]: I think that's a good idea too. Done.

**Comment by hcallahan-lowrisc** [comment]
Did a quick run-through of the grammar in the last push. Thanks @rswarbrick

**Review by rswarbrick — APPROVED** [review]
Looks good to me!

**Comment by rswarbrick** [comment]
CHANGE AUTHORIZED: hw/ip/spi_device/data/spi_device.hjson

**Review by marnovandermaas — APPROVED** [review]
Looks reasonable to me, let's get this merged and we can always iterate later.

**Comment by marnovandermaas** [comment]
CHANGE AUTHORIZED: hw/ip/spi_device/data/spi_device.hjson

**Pull request merged**